app.js

```diff
@@ -68,6 +68,222 @@ function nonce( length ) {
     return text;
 }
 
+/**
+ * 
+ * @param { string } password 
+ * @returns { string | null }
+ */
+async function fetchClientIdIfValidAsync( password ) {
+  const requiredScopes = [ "channel:read:redemptions", "user:read:email" ];
```

Comment:
Should we make this optionally definable so that it can support more than redemptions too?